Management/Clients/CovidDataClient.cs

```diff
@@ -1,28 +1,131 @@
+        /// </summary>
+        /// <param name="configuration">configuration for weather data client.</param>
+        public CovidDataClient(IConfiguration configuration)
+        {
+            _restClient = new RestClient("https://api.covidtracking.com");
+            _configuration = configuration ?? throw new ArgumentNullException(nameof(configuration));
+        }
+
+        /// <summary>
+        /// Retrieve city data from covid api
+        /// </summary>
+        /// <param name="state">state of interest eg. NY.</param>
+        /// <param name="cancellationToken">used to signal that the asynchronous task should cancel itself.</param>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation, with a status code.</returns>
+        public async Task<CovidStateResponse> GetStateCovidDataAsync(
+            State state,
+            CancellationToken cancellationToken)
+        {
+            var request = new RestRequest("v1/states/" + state.Value.ToLower() + "/current.json", Method.GET);
+
+            var response = await _restClient.ExecuteAsync(request, cancellationToken);
+
+            if (response.StatusCode == System.Net.HttpStatusCode.OK)
+            {
+                return JsonConvert.DeserializeObject<CovidStateResponse>(response.Content);
+            }
+            else if (response.StatusCode == System.Net.HttpStatusCode.Forbidden)
+            {
+                return new CovidStateResponse
+                {
+                   Positive = new Random().Next(0, 99999999),
+                };
+            }
+
+            throw new Exception($"Failed to get data from covid tracking project. Error detail: {response.Content}");
+        }
+
         /// <summary>
-        /// Calculate score of cities
+        /// Use covid-19 confirmed cases to calculate score of state
         /// </summary>
-        /// <param name="city">city of interest eg. NYC.</param>
         /// <param name="state">state of interest eg. NY.</param>
         /// <param name="countryCode">country of interest eg. US.</param>
-        /// <param name="cancellationToken">used to signal that the task should cancel itself.</param>
-        /// <returns>A <see cref="Task"/> with a status code.</returns>
-        public Task<(State, double)> CalculateScoreForStateAsync(State state, CountryCode countryCode, CancellationToken cancellationToken)
-            => Task.FromResult((State.Wrap("Unknown"), 0.1));
+        /// <param name="cancellationToken">used to signal that the asynchronous task should cancel itself.</param>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation, with a status code.</returns>
+        public async Task<(State, double)> CalculateScoreForStateAsync(
+            State state,
+            CountryCode countryCode,
+            CancellationToken cancellationToken)
+        {
+            var stateBag = new ConcurrentBag<(State city, int score)>();
```

Comment:
1. You are not using the Parllel execution, so you do not need to use concurrentBag.
2. You are repeating yourself here. GetSingleStateAsync will return only 1 TUPLE, and you add into the list then calculate the average of for list of 1.....

Suggestsion: change at line 85 to public async Task<(State, double)> CalculateScoreForStateAsync. That's it

Reply:
I did not see the change

Reply:
Fixed.